agent/log-drop/log-drop.go

```diff
@@ -0,0 +1,70 @@
+package log_drop
+
+import (
+	"context"
+)
+
+type Level int
+
+const (
+	TRACE Level = iota
+	DEBUG
+	INFO
+	WARN
+	ERROR
+)
+
+//go:generate mockery --name Logger --inpackage
+type Logger interface {
+	Info(string, ...interface{})
+}
```

Comment:
What do you think of implementing the [`hclog.Logger` interface](https://pkg.go.dev/github.com/hashicorp/go-hclog#Logger) to make it easier to slot into places where we already depend on it?

Reply:
I thought about that but I think implementing the full interface is a bit overkilled for what we need for now. At the same time I see a use for this kind of logger in places where we would like to add some debug logging without adding too much performance hit 🤷

Reply:
Yeah, I think it'd be ok to embed the wrapped logger and only implement the methods we need:

```Go
var _ hclog.Logger = (*Logger)(nil)

type Logger struct {
  hclog.Logger
}

// We just implement these methods.
func (l *Logger) Info(s string, i ...interface{}) {
  r.pushLog(Log{l: INFO, s: s, i: i})
}

func (l *Logger) Debug(s string, i ...interface{}) {
  r.pushLog(Log{l: DEBUG, s: s, i: i})
}

// Everything else gets called on the base logger.
```

Reply:
I think I have an idea. I can implement a log Sink instead. Which would give us the opportunity to use the same log interceptor we usually use and make it apply to all hclog.Logger interface. 

I should have something around that in the PR shortly 😄